grader/src/main/java/edu/pdx/cs410J/grader/scoring/ProjectSubmissionOutputFileParser.java

```diff
@@ -1,27 +1,106 @@
 package edu.pdx.cs410J.grader.scoring;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.Reader;
+import java.io.StringReader;
+import java.nio.CharBuffer;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
 import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Pattern;
 
 /**
  * Author: Susham Kumar
  */
 
 public class ProjectSubmissionOutputFileParser {
-  public ProjectSubmissionOutputFileParser(Reader source) {
 
+  public ProjectSubmissionOutputFileParser(Reader source) {
+      if(source != null) {
+          bufferedReader = new BufferedReader(source);
+      }
 
   }
 
-  public ProjectSubmission parse() {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ProjectSubmission parse()  {
+
+     projectSubmission = new ProjectSubmission();
+     String fileContents=null;
+
+      try {
+          while((fileContents= bufferedReader.readLine()) !=null) {
+            fileContents=fileContents.trim();
+          if(fileContents.contains("CS410J")){              // fileContents has the student Id and Project Name.
+              String[] splitContent=fileContents.split(Pattern.quote("."));
+              projectSubmission.setProjectName(splitContent[splitContent.length -1]);
+              projectSubmission.setStudentId(splitContent[splitContent.length -2]);
+
+            }
+
+            if(fileContents.contains("Submitted by")){ //fileContents has StudentName
+             String [] splitContent=fileContents.split("Submitted by ");
+             projectSubmission.setStudentName(splitContent[splitContent.length -1]);
+            }
+
+            if(fileContents.contains("Submitted on")){ //fileContents has Submission Time
+                String [] splitContent=fileContents.split("Submitted on ");
+                projectSubmission.setSubmissionTime(parseSubmissionTime(splitContent[splitContent.length -1]));
+
+            }
+
+            if(fileContents.contains("Graded on")){ //fileContents has Graded Time
+                String [] splitContent=fileContents.split("Graded on");
+                projectSubmission.setGradedTime(parseGradingTime(splitContent[splitContent.length -1].trim()));
+            }
+              if(fileContents.contains("out of")){ //fileContents has Total Points and Awarded Points
+                  String [] splitContent=fileContents.split("out of");
```

Comment:
This works, but you might also want to consider using two "capturing groups" in a regular expression:

https://alvinalexander.com/blog/post/java/how-extract-multiple-groups-patterns-string-regex

Reply:
Is it possible to create a regexp to get the left and right side of the matched string? Right now, I am able to get both the left and right sub string of the matched string using a separate Regexp